src/resources/embeddings.ts

```diff
@@ -9,9 +9,53 @@ export class Embeddings extends APIResource {
    */
   create(
     body: EmbeddingCreateParams,
-    options?: Core.RequestOptions,
+    options?: Core.RequestOptions<EmbeddingCreateParams>,
   ): Core.APIPromise<CreateEmbeddingResponse> {
-    return this._client.post('/embeddings', { body, ...options });
+    Core.debug('request', 'Sending request with arguments:', { body, ...options });
+
+    const hasUserProvidedEncodingFormat = !!body.encoding_format;
+    let encoding_format: EmbeddingCreateParams['encoding_format'] =
+      hasUserProvidedEncodingFormat ? body.encoding_format : 'base64';
+
+    if (hasUserProvidedEncodingFormat) {
+      Core.debug('Request', 'User defined encoding_format:', body.encoding_format);
+    } else {
+      // No encoding_format specified, defaulting to base64 for performance reasons
+      // See https://github.com/openai/openai-node/pull/1312
+      encoding_format = 'base64';
+    }
```

Comment:
nit: this `else` block is redundant because you already default to `base64` on line 18, comment is useful though so can you move the comment to the assignment above?

Reply:
fixed.